libs/src/services/discoveryProvider/index.js

```diff
@@ -544,19 +544,29 @@ class DiscoveryProvider {
 
     try {
       const response = await axios(axiosRequest)
-      const {
-        data,
-        latest_indexed_block: indexedBlock,
-        latest_chain_block: chainBlock
-      } = Utils.parseDataFromResponse(response)
-
-      if ((chainBlock - indexedBlock) > UNHEALTHY_BLOCK_DIFF) {
-        const endpoint = await this.autoSelectEndpoint()
-        this.setEndpoint(endpoint)
-        throw new Error(`Selected endpoint was too far behind. Indexed: ${indexedBlock} Chain: ${chainBlock}`)
+      const parsedResponse = Utils.parseDataFromResponse(response)
+
+      if (
+        'latest_indexed_block' in parsedResponse &&
+        'latest_chain_block' in parsedResponse
+      ) {
+        const {
+          latest_indexed_block: indexedBlock,
+          latest_chain_block: chainBlock
+        } = parsedResponse
+
+        if (
+          !chainBlock ||
+          !indexedBlock ||
+          (chainBlock - indexedBlock) > UNHEALTHY_BLOCK_DIFF
+        ) {
+          const endpoint = await this.autoSelectEndpoint()
+          this.setEndpoint(endpoint)
+          throw new Error(`Selected endpoint was too far behind. Indexed: ${indexedBlock} Chain: ${chainBlock}`)
+        }
       }
 
-      return data
+      return parsedResponse.data
     } catch (e) {
       console.error(e)
```

Comment:
3 retries makes sense when querying against a single discprov but now that we are rotating, does it still? for example, if you catch the Error from above of "selected endpoint was too far behind" then might it make sense to re-set the retry counter? we can't do exactly this since we would then potentially have infinite loop of re-selection + 3 retries, so what about just increasing this counter to be more lenient in the short term, something like 5-7?